openproblems/tasks/spatial_decomposition/methods/vanillanmf.py

```diff
@@ -0,0 +1,67 @@
+from ....tools.decorators import method
+from ....tools.normalize import log_cpm
+from ....tools.utils import check_version
+from sklearn.decomposition import NMF
+
+import numpy as np
+import pandas as pd
+
+
+@method(
+    method_name="Non-Negative Matrix Factorization (NMF).",
+    paper_name="Fast local algorithms for large scale nonnegative matrix and tensor factorizations",
+    paper_url="https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.214.6398&rep=rep1&type=pdf",
+    paper_year=2009,
+    code_url="https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.NMF.html",
+    code_version=check_version("scikit-learn"),
+)
+# Used from https://github.com/theislab/scanpy/issues/181#issuecomment-534867254
+def grouped_obs_mean(adata, group_key, layer=None):
```

Comment:
I have the same function in my branch, when both are merged, let's settle on one and use that in all cases where means over cell types are needed.